src/main.ts

```diff
@@ -0,0 +1,242 @@
+      cmd.push('--gcs-source-staging-dir', sourceStagingDir);
+    }
+    if (skaffoldFile) {
+      cmd.push('--skaffold-file', skaffoldFile);
+    }
+
+    const allAnnotations = Object.assign({}, getDefaultAnnotations(), annotations);
+    cmd.push('--annotations', kvToString(allAnnotations));
+
+    const allLabels = Object.assign({}, getDefaultLabels(), labels);
+    cmd.push('--labels', kvToString(allLabels));
+
+    if (description) {
+      cmd.push('--description', description);
+    }
+    if (flags) {
+      const flagList = parseFlags(flags);
+      if (flagList) {
+        cmd.push('--flags');
+        cmd = cmd.concat(flagList);
+      }
+    }
+
+    // Set output format to json for easy parsing
+    cmd.push('--format', 'json');
+
+    // Install gcloud if not already installed.
+    if (!isGcloudInstalled(gcloudVersion)) {
+      await installGcloudSDK(gcloudVersion);
+    } else {
+      const toolPath = toolCache.find('gcloud', gcloudVersion);
+      addPath(path.join(toolPath, 'bin'));
+    }
+
+    // Install gcloud component if needed and prepend the command
+    if (gcloudComponent) {
+      await installGcloudComponent(gcloudComponent);
+      cmd.unshift(gcloudComponent);
+    }
+
+    // Authenticate - this comes from google-github-actions/auth.
+    const credFile = process.env.GOOGLE_GHA_CREDS_PATH;
+    if (credFile) {
+      await authenticateGcloudSDK(credFile);
+      logInfo('Successfully authenticated');
+    } else {
+      logWarning('No authentication found, authenticate with `google-github-actions/auth`.');
+    }
+
+    const toolCommand = getToolCommand();
+    const options = { silent: true, ignoreReturnCode: true };
+    const commandString = `${toolCommand} ${cmd.join(' ')}`;
+    logInfo(`Running: ${commandString}`);
+
+    // Run gcloud cmd
+    const output = await getExecOutput(toolCommand, cmd, options);
+    if (output.exitCode !== 0) {
+      const errMsg = output.stderr || `command exited ${output.exitCode}, but stderr had no output`;
+      throw new Error(`failed to execute gcloud command \`${commandString}\`: ${errMsg}`);
+    }
+
+    const outputs: CreateCloudDeployReleaseOutputs = parseCreateReleaseResponse(output.stdout);
+
+    if (outputs.name) {
+      setOutput('name', outputs.name);
+    }
+    if (outputs.link) {
+      setOutput('link', outputs.link);
+    }
+  } catch (err) {
+    const msg = errorMessage(err);
+    setFailed(`create-cloud-deploy-release failed with: ${msg}`);
+  } finally {
+    restoreEnv();
+  }
+}
+
+/**
+ * Convert string=string records into a single string for the gcloud CLI
+ */
+export function kvToString(kv: Record<string, string>, separator = ','): string {
+  return Object.entries(kv)
+    .map(([k, v]) => {
+      return `${k}=${v}`;
+    })
+    .join(separator);
+}
+
+/**
+ * computeGcloudVersion computes the appropriate gcloud version for the given
+ * string.
+ */
+async function computeGcloudVersion(str: string): Promise<string> {
```

Comment:
This should not be required anymore. See https://github.com/google-github-actions/setup-cloud-sdk/commit/747b74e47b684adc0223f0371d4a159abb42cfb6